archimedes/meta_obj.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2014-2018 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#   Valerio Cosentino <valcos@bitergia.com>
+#
+
+import json
+
+
+class MetaObj:
+    """This class models the metadata information of Kibana objects which includes the ID, title, type and version
+    of the object plus when the object was updated if not null.
+
+    :param _id: object id
+    :param _title: object title
+    :param _type: object type
+    :param _version: object version
+    :param _updated_at: time when the object was updated, it can be null
+    """
+    def __init__(self, _id, _title, _type, _version, _updated_at=None):
+        self.id = _id
+        self.title = _title
+        self.type = _type
+        self.version = _version
+        self.updated_at = _updated_at
+
+    def __repr__(self):
+        meta = {
+            'id': self.id,
+            'type': self.type,
+            'version': self.version,
+            'title': self.title
+        }
+
+        if self.updated_at:
+            meta['updated_at'] = self.updated_at
+
+        return json.dumps(meta, sort_keys=True, indent=4)
+
+    @classmethod
+    def create_from_obj(cls, obj):
+        """Create a MetaObj from a Kibana object
+
+        :param obj: Kibana object
+        """
+        updated_at = obj['updated_at'] if 'updated_at' in obj else None
+
+        return MetaObj(obj['id'], obj['attributes']['title'], obj['type'], obj['version'], updated_at)
+
+    @classmethod
+    def create_from_registry(cls, entry):
+        """Create a MetaObj from a entry in the registry
+
+        :param entry: registry entry
```

Comment:
Is this a registry entry? It looks like a dictionary. Probably should be good to have objects to store these entries.

Reply:
A Registry entry is a JSON object, at the moment there is not Python class modeling an entry. I can add it (same thing for Kibana objects) in another PR.